src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -139,6 +139,30 @@
          [Spinner {:text (:banner props)}]]]))})
 
 
+(react/defc SafeBlocker
+  {:show
+   (fn [{:keys [props state]}]
+     (swap! state assoc :show-requested? true)
+     (js/setTimeout #(when (:show-requested? @state)
+                      (swap! state assoc :showing? true))
+                    (:delay-time props)))
+   :hide
+   (fn [{:keys [state]}]
+     (swap! state dissoc :showing? :show-requested?))
+   :get-default-props
+   (fn []
+     {:delay-time 200})
+   :render
+   (fn [{:keys [props state]}]
+     (when (:showing? @state)
+       [:div {:style {:backgroundColor "rgba(210, 210, 210, 0.4)"
+                      :position "absolute" :top 0 :bottom 0 :right 0 :left 0 :zIndex 9999}}
+        [:div {:style {:position "absolute" :top "50%" :left "50%"
+                       :transform "translate(-50%, -50%)"
```

Comment:
can you explain this? looks like the top/left and transform cancel each other out.